### ISSUE TITLE
Why View::head does not trigger event

### ISSUE THREAD

**Issue by yuniorsk** [yiisoft/yii2#17401]
Hi, is there any particular reason why `View::head` method does not trigger event like other such methods (beginBody, beginEnd etc.)?

**Comment by samdark** [comment]
What's the use case for adding it?

**Comment by yuniorsk** [comment]
Hi, I am using `View::EVENT_BEGIN_BODY` and `View::EVENT_END_BODY` to inject some custom scripts into template and I would like to use same principle for head. I was also considering using `View::register*` methods for this but it's not that flexible, e.g. I ran into problem how to render `<noscript>` tag using those methods.

**Comment by samdark** [comment]
Have you tried using `View::EVENT_BEGIN_PAGE` or `View::EVENT_BEFORE_RENDER`?

**Comment by yuniorsk** [comment]
I am not sure how would I do that. How would you inject custom html into head using those?

Here a snippet of what I am trying to accomplish. I am sure there's another way how to do it, but I was just curious if there any particular reason why `yii\web\View::head` does not trigger event just like its counterparts.

https://gist.github.com/yuniorsk/502f00cd20129ccaf766b73c54bce085

**Comment by samdark** [comment]
Ah, got it. One last thing: https://github.com/yiisoft/yii2/blob/c8564241d3cd3afa09077e043472957a8479791f/framework/helpers/BaseHtml.php#L226

**Comment by yuniorsk** [comment]
Yep, that could work but that event would be still more convenient in my case because it would be handled in same unified way rather than handling head section with some different logic.

**Comment by samdark** [comment]
While it makes sense to have it, Yii 2 isn't accepting enhancements anymore so I'll move this one to 3.0.
